src/lib/bootloader/systemdboot_widgets.rb

```diff
@@ -105,6 +106,38 @@ def store
       end
     end
 
+    # Represents decision if smt is enabled
+    class SdCpuMitigationsWidget < CpuMitigationsWidget
+      include SystemdBootHelper
+
+      def init
+        if systemdboot.respond_to?(:cpu_mitigations)
+          self.value = systemdboot.cpu_mitigations.value.to_s
+        else
+          # do not crash when use no bootloader. This widget is also used in security dialog.
+          # (bsc#1184968)
```

Comment:
yeah and this shows that it can lead to potential issues with naming as new name is used in security dialog. So renaming security_boot is problematic and can be even more when we do not know if it is systemd boot or grub2. And also code is almost identical. So what I proper here is to not have divided widgets  which differes just in name of bootloader nad instead of including that helpers define something like:
```
def settings
  case BootloaderFactory.current
  when Grub2Base then BootloaderFactory.current.grub_default
  when SystemdBoot then BootloaderFactory.current
  else
     nil
  end
end
```
and in code have something like
```
def init
  if settings
    self.value =settings.cpu_mitigations.value.to_s
  else
    disable
  end
end
```

and similar for rest of code. Basically allow widget to work with both dynamically instead of copying code here and there.
```

```

Reply:
Hm, I have not managed to go exactly that way, but I hope that I have mainly hit your requirements.